### ISSUE TITLE
New cop to detect unnecessary `merge` when passing keyword arguments

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#13252]
```ruby
# bad
some_method(foo, **options.merge(bar: 1))

# good
some_method(foo, **options, bar: 1)
```

I found (grep by `\*\*\w+\.merge` regexp) 12 such offences in the `rails/rails` codebase and 14 in my own gem (https://github.com/fatkodima/online_migrations).

**Comment by koic** [comment]
Hm. Whether to use `merge` with a single argument or separate it may come down to personal preference, so it seems reasonable to propose this to the style guide first. Since it may involve a trade-off in expressiveness, I’m not sure yet if it’s best to enforce one way or the other.

**Comment by vlad-pisanov** [comment]
The `merge` option is a lot slower on my machine, especially when building new hashes. This could be a candidate for `rubocop-performance` as well

```ruby
require 'benchmark/ips'

data = {}

Benchmark.ips do |x|
  x.report('merge') { { **data.merge(foo: 42) } }
  x.report('**')    { { **data, foo: 42 } }

  x.compare!
end

```

```sh
ruby 3.3.2 (2024-05-30 revision e5a195edf6) [x86_64-darwin21]
Warming up --------------------------------------
               merge   256.750k i/100ms
                  **   585.002k i/100ms
Calculating -------------------------------------
               merge      2.490M (± 5.8%) i/s -     12.581M in   5.069580s
                  **      6.010M (± 8.6%) i/s -     29.835M in   5.005139s

Comparison:
                  **:  6009715.4 i/s
               merge:  2489574.5 i/s - 2.41x  slower
```

**Comment by fatkodima** [comment]
It is slower and I wonder why people might prefer to use `merge` after knowing that it is really not needed.

**Comment by koic** [comment]
I see. In any case, it would be worth documenting this in the style guide.

**Comment by dvandersluis** [comment]
I opened https://github.com/rubocop/ruby-style-guide/pull/948 for this.
